src/datasources/perf-ds/datasource.ts

```diff
@@ -341,6 +343,29 @@ export class OpenNMSDatasource {
     return Promise.resolve([]);
   }
 
+  metricFindNodesLocation(searchLimit) {
+    return this.doOpenNMSRequest({
+      url: '/rest/nodes',
+      method: 'GET',
+      params: {
+        limit: searchLimit ? searchLimit : 0        
+      }
+    })
+    .then(function(response){
+      if (response.data.count > response.data.totalCount) {
+        console.warn("Filter matches " + response.data.totalCount + " records, but only " + response.data.count + " will be used.");
+      }
+      var results = [] as any[];
+      _.each(response.data.node, function (node) {
+        var nodeLocation = node.location.toString();
+        if(nodeLocation){
+        results.push({text: nodeLocation, value: nodeLocation, expandable: true});
+        }
+      });
+      return results;
+    });
```

Comment:
This seems to do the wrong thing. There is a one-to-many relation between location and node. Therefore this would add the same location multiple times if multiple nodes are assigned to the same location.

Reply:
I was goint to filter but grafana already does it. I can pass 200 'Default' locations and only one shows up.

Still it is confusing, so I'll change it.

Reply:
switechd to rest monitoringLocations which seems to hae unique values, but added a check to ensure no duplicates are passed.